crypto/src/main/scala/coop/rchain/crypto/util/Sorting.scala

```diff
@@ -0,0 +1,13 @@
+package coop.rchain.crypto.util
+
+import coop.rchain.crypto.PublicKey
+import coop.rchain.crypto.codec.Base16
+
+object Sorting {
+
+  implicit val byteArrayOrdering: Ordering[Array[Byte]] =
+    Ordering.by(Base16.encode)
+
+  implicit val publicKeyOrdering: Ordering[PublicKey] =
+    (x: PublicKey, y: PublicKey) => byteArrayOrdering.compare(x.bytes, y.bytes)
```

Comment:
Since Ordering is a ContravariantFunctor, you can make this better like this:

```scala
  import cats.implicits._ //waaaay up in the imports

  val o: Ordering[PublicKey] = Ordering.by(Base16.encode).contramap(_.bytes)
```

Besides satisfying one's FP kink, we also get to not call `.bytes` twice (which begs for an error!) and have no way of swapping the args involuntarily.

:sunglasses:

Reply:
👍